controllers/virtualmachinegroup/virtualmachinegroup_controller.go

```diff
@@ -593,28 +581,58 @@ func (r *Reconciler) getPlacementMembers(
 func (r *Reconciler) getVMForPlacement(
 	ctx context.Context,
 	vmGroup *vmopv1.VirtualMachineGroup,
-	vmName string) (*vmopv1.VirtualMachine, error) {
+	vmName string) (vm *vmopv1.VirtualMachine, err error) {
 
-	if _, member := findMemberStatus(vmName, vmKind, vmGroup.Status.Members); member != nil {
-		if conditions.IsTrue(member, vmopv1.VirtualMachineGroupMemberConditionPlacementReady) {
-			return nil, nil
-		}
-		if !conditions.IsTrue(member, vmopv1.VirtualMachineGroupMemberConditionGroupLinked) {
-			return nil, fmt.Errorf("VM %q is not linked for group %q", vmName, vmGroup.Name)
-		}
-	} else {
+	_, memberStatus := findMemberStatus(vmName, vmKind, vmGroup.Status.Members)
+	if memberStatus == nil {
 		return nil, fmt.Errorf("VM %q is not in group member status", vmName)
 	}
 
-	vm := &vmopv1.VirtualMachine{}
-	if err := r.Get(ctx, client.ObjectKey{Name: vmName, Namespace: vmGroup.Namespace}, vm); err != nil {
+	defer func() {
+		if err != nil {
+			conditions.MarkError(
+				memberStatus,
+				vmopv1.VirtualMachineGroupMemberConditionPlacementReady,
+				"Error",
+				err)
+		} else if vm == nil {
+			// If both error and vm are nil, it means the VM is already placed,
+			// or will be placed outside the group. Update the PlacementReady
+			// condition to True to be able to set the group as ready.
+			conditions.MarkTrue(
```

Comment:
For PlacementReady=true in these situations I thought we had agreed to include some extra details (like Reason=AlreadyPlaced) for extra details about why the placement status is otherwise empty?

Reply:
Moved the VM.status.uniqueID check first to update the PlacementReady with AlreadyPlaced correctly.